enterprise/server/githubapp/githubapp.go

```diff
@@ -414,10 +416,45 @@ func (a *GitHubApp) maybeTriggerBuildBuddyWorkflow(ctx context.Context, eventTyp
 	if err != nil {
 		return err
 	}
+	if err := a.MaybeReindexRepo(ctx, row.GitRepository, wd, repoURL, tok.GetToken()); err != nil {
+		log.Debugf("Not indexing repo: %s", err)
+	}
 	return a.env.GetWorkflowService().HandleRepositoryEvent(
 		ctx, row.GitRepository, wd, tok.GetToken())
 }
 
+func (a *GitHubApp) MaybeReindexRepo(ctx context.Context, repo *tables.GitRepository, wd *interfaces.WebhookData, repoURL *gitutil.RepoURL, accessToken string) error {
+	codesearchService := a.env.GetCodesearchService()
+	if codesearchService == nil {
+		return nil
+	}
+	if wd.PushedBranch != "main" && wd.PushedBranch != "master" {
+		return nil
+	}
+	g, err := a.env.GetUserDB().GetGroupByID(ctx, repo.GroupID)
+	if err != nil {
+		return err
+	}
+	if !g.CodeSearchEnabled {
+		return nil
+	}
+	_, err = codesearchService.Index(ctx, &csinpb.IndexRequest{
+		GitRepo: &gitpb.GitRepo{
+			RepoUrl:     repoURL.String(),
+			AccessToken: accessToken,
```

Comment:
You aren't using this access token anywhere - you'll have to set it on the repo url to access private repos

(And so I don't forget later - if you do add the token to the repo url, make sure not to log it)

Reply:
Yup, I am planning to do this in a followup because this CL was already getting too big!